engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobRecurProcessor.java

```diff
@@ -37,32 +39,19 @@ public boolean onCommand(
     final long jobKey = command.getKey();
     final JobState.State state = jobState.getState(jobKey);
 
-    if (state == State.FAILED) {
-      final JobRecord recurredJob = command.getValue();
+    preconditionChecker
+        .check(state, jobKey)
+        .ifRightOrLeft(
+            ok -> {
+              final JobRecord recurredJob = command.getValue();
 
-      commandControl.accept(JobIntent.RECURRED_AFTER_BACKOFF, recurredJob);
-      jobActivationBehavior.publishWork(recurredJob);
-    } else {
-      final String textState;
+              commandControl.accept(JobIntent.RECURRED_AFTER_BACKOFF, recurredJob);
+              jobActivationBehavior.publishWork(recurredJob);
+            },
+            violation -> {
+              commandControl.reject(violation.getLeft(), violation.getRight());
+            });
 
-      switch (state) {
-        case ACTIVATABLE:
-          textState = "it is already activable";
-          break;
-        case ACTIVATED:
-          textState = "it is already activated";
-          break;
-        case ERROR_THROWN:
-          textState = "it is in error state";
-          break;
-        default:
-          textState = "no such job was found";
-          break;
-      }
```

Comment:
:x: Maybe I missed something, but it looks like with your changes, the error messages are now changed, i.e. we only provide a `NO_JOB_FOUND_MESSAGE` or `INVALID_JOB_STATE_MESSAGE`.

With your changes, we will make things more consistent, but our customers may rely on the existing error messages, and they may have introduced some automation on them.

I would suggest to find a way to keep the error messages as they are.